src/app/clusters/ethernet-network-diagnostics-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,96 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/clusters/ethernet-network-diagnostics-server/ethernet-diagnostics-cluster.h>
+#include <app/static-cluster-config/EthernetNetworkDiagnostics.h>
+#include <app/util/attribute-storage.h>
+#include <data-model-providers/codegen/CodegenDataModelProvider.h>
+#include <platform/DiagnosticDataProvider.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::EthernetNetworkDiagnostics;
+
+// the static assert is skipped in case of dynamic endpoints.
+static_assert((EthernetNetworkDiagnostics::StaticApplicationConfig::kFixedClusterConfig.size() == 1 &&
+               EthernetNetworkDiagnostics::StaticApplicationConfig::kFixedClusterConfig[0].endpointNumber == kRootEndpointId) ||
+              EthernetNetworkDiagnostics::StaticApplicationConfig::kFixedClusterConfig.size() == 0);
+
+namespace {
+
+LazyRegisteredServerCluster<EthernetDiagnosticsServerCluster> gServer;
+
+// compile-time evaluated method if "is <EP>::EthernetNetworkDiagnostics::<ATTR>" enabled
+constexpr bool IsAttributeEnabled(EndpointId endpointId, AttributeId attributeId)
+{
+    for (auto & config : EthernetNetworkDiagnostics::StaticApplicationConfig::kFixedClusterConfig)
+    {
+        if (config.endpointNumber != endpointId)
+        {
+            continue;
+        }
+        for (auto & attr : config.enabledAttributes)
+        {
+            if (attr == attributeId)
+            {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+} // namespace
+
+void emberAfEthernetNetworkDiagnosticsClusterInitCallback(EndpointId endpointId)
+{
+    VerifyOrReturn(endpointId == kRootEndpointId);
+
+    const EthernetDiagnosticsEnabledAttributes enabledAttributes{
+        .enableCarrierDetect  = IsAttributeEnabled(kRootEndpointId, Attributes::CarrierDetect::Id),
+        .enableFullDuplex     = IsAttributeEnabled(kRootEndpointId, Attributes::FullDuplex::Id),
+        .enablePacketCount    = IsAttributeEnabled(kRootEndpointId, Attributes::PacketRxCount::Id),
+        .enablePHYRate        = IsAttributeEnabled(kRootEndpointId, Attributes::PHYRate::Id),
+        .enableTimeSinceReset = IsAttributeEnabled(kRootEndpointId, Attributes::TimeSinceReset::Id),
+        .enableErrCount       = IsAttributeEnabled(kRootEndpointId, Attributes::TxErrCount::Id),
+    };
+
+    gServer.Create(DeviceLayer::GetDiagnosticDataProvider(), enabledAttributes);
```

Comment:
so even though we are supposed to support secondary network interfaces, we have a single "diagnostics provider"? It does seem like we do not properly support secondary network interfaces :( Pre-existing but maybe we should add a comment and issues... the diagnostics should depend on the endpointId probably ....